tests/src/tests_pointerhash.pas

```diff
@@ -420,41 +420,36 @@ procedure TTestsPointerHashList.TestAddDuplicateError;
   psinvl^ := 'value3.2';
 
   try
-    if Self.lsthshobjs.Add('key3', psinvl) then
-    begin
-      //The Value was unexpectly added and the former Value must be freed
-      Dispose(psoutvl);
+    Self.lsthshobjs.Add('key3', psinvl);
+    //This is a faulty Behaviour with "dupError"
+    Fail('INS - TPLPointerHashList.Add() failed! Duplicate Key was accepted.');
+  except
+    on e: DuplicateKeyException do
+      begin
+        writeln('Duplicate hash key detected. ' + e.Message);
```

Comment:
I added this and the next writeln() simply to for debugging purposes to show which exception handler is called.